### PR TITLE
Do not overwrite the shareType so the post hook is still correct

### PR DESCRIPTION
**Pull request by nickvergessen**
@LukasReschke fixes missing group share activities in activity app

cc @schiesbn please review aswell

@DeepDiver1975 @karlitschek please approve backport to stable8 and stable7 :/

### CODE REVIEW

**Comment by schiesbn** [comment]
looks good :+1:

**Comment by karlitschek** [comment]
:+1:

File: lib/private/share/share.php [hunk 1]
```diff
@@ -1850,7 +1850,7 @@ private static function put($itemType, $itemSource, $shareType, $shareWith, $uid
 			$sourceId = ($itemType === 'file' || $itemType === 'folder') ? $fileSource : $itemSource;
 			$sourceExists = self::getItemSharedWithBySource($itemType, $sourceId, self::FORMAT_NONE, null, true, $user);
 
-			$shareType = ($isGroupShare) ? self::$shareTypeGroupUserUnique : $shareType;
+			$userShareType = ($isGroupShare) ? self::$shareTypeGroupUserUnique : $shareType;
```
Comment (PVince81) [inline lib/private/share/share.php]: If I understand well the bug was that the `$shareType` variable was overwriting another one, so you just renamed it ?

Comment (nickvergessen) [inline lib/private/share/share.php]: Yes, `$shareType` is the parameter of the method, while this in here was a "local variable"
The overwriting was unintentional and only affects the `post_shared` hook

Comment (MorrisJobke) [inline lib/private/share/share.php]: global variables magic ;)

**Comment by PVince81** [comment]
:+1:

**Comment by PVince81** [comment]
I should have tested before merging this...

Let's at least test for stable7. @nickvergessen I need the test steps. See https://github.com/owncloud/core/pull/13988#issuecomment-73889478

**Comment by MorrisJobke** [comment]
stable7 #13988
stable8 #13987

**Pull request merged**